src/controllers/token.ts

```diff
@@ -31,6 +31,36 @@ export default class TokenController {
     }
   }
 
+  @Post('/waitlist')
+  async addWaitlistToken(
+    @Body({ required: true }) body: TokenWithSettings,
+    @Ctx() ctx: Context
+  ) {
+    const { token } = body
+    const existingToken = await TokenModel.findOne({ token })
+    if (existingToken) {
+      existingToken.waitlist = true
+      await existingToken.save()
+      return { success: true }
+    }
+    if (!token) return ctx.throw(badRequest('No token'))
+    const newToken = new TokenModel({ token, waitlist: true })
+    await newToken.save()
+    return { success: true }
+  }
+
+  @Put('/waitlist/:token')
+  async markAsReceived(
+    @Params('token') notificationToken: string,
```

Comment:
I get the idea, but seems like it's better to name it `token`, instead of `notificationToken` 🤔